task_processing/interfaces/event.py

```diff
@@ -8,18 +8,36 @@
 
 
 class Event(PRecord):
-    timestamp = field(type=datetime.datetime)
+    type = field(type=str, initial=None)
```

Comment:
that was my original thinking, but considering that events are really just dicts and our type system is there purely for convenience, it is worth having the dict to be self-descriptive rather than relying on type being attached to it everywhere. we would lose that type information during serialization for instance.